### PR TITLE
Add Job Guard to each integration job script

### PR DESCRIPTION
**Pull request by mszostok**
**Description**

Changes proposed in this pull request:

- Add Job Guard to each integration job script
- Fix the Job Guard:
  - check only those jobs which should be reported. Previously there was a problem if pr has reported more jobs, e.g. optional job was triggered manually so status was added to checks.
  ```
  2019/05/31 13:57:35 Got 227 statuses, expected 226
  ```
  - add body msg when unexpected status was returned
  ```
  2019/05/31 14:01:04 returned unexpected status code, expected: [200], got: [403]
  ```

### Tests

CI logs when some component job is failed:
- https://storage.googleapis.com/kyma-prow-logs/pr-logs/pull/kyma-project_kyma/4191/pre-master-kyma-gke-integration/1135486026584690688/build-log.txt

CI execution when all components are success:
- https://storage.googleapis.com/kyma-prow-logs/pr-logs/pull/kyma-project_kyma/4191/pre-master-kyma-gke-integration/1135198757877452800/build-log.txt

**Related issue(s)**

- #978

### CODE REVIEW

File: development/tools/cmd/jobguard/main.go [hunk 1]
```diff
@@ -39,63 +39,67 @@ func main() {
...
-		if pendingStatusesLen > 0 {
-			log.Printf("Waiting for jobs to finish:\n%s", printJobNames(pendingStatuses))
+		switch {
+		case len(notYetReported) > 0:
+			log.Printf("Waiting for jobs [%s] to report their status", printJobNames(notYetReported))
 			return false, nil
+		case len(pending) > 0:
+			log.Printf("Waiting for jobs to finish: [%s]", printJobNames(pending))
+			return false, nil
+		case len(failed) > 0:
```
Comment (aszecowka) [inline development/tools/cmd/jobguard/main.go]: move `case len(failed) > 0` as a first one in the `switch` statement

File: development/tools/jobs/kyma/kyma_integration_test.go [hunk 2]
```diff
@@ -305,57 +300,50 @@ func TestKymaGKECentralConnectorJobsPresubmit(t *testing.T) {
...
-
-			expJobImage: tester.ImageBootstrap001,
 		},
 		"Should contains the gke-integration job": {
 			givenJobName: "post-master-kyma-gke-integration",
 
 			expPresets: []tester.Preset{
 				tester.PresetGCProjectEnv, tester.PresetBuildMaster,
-				tester.PresetDindEnabled, "preset-sa-gke-kyma-integration",
+				tester.PresetDindEnabled, tester.PresetKymaGuardBotGithubToken, "preset-sa-gke-kyma-integration",
```
Comment (aszecowka) [inline development/tools/jobs/kyma/kyma_integration_test.go]: what is the purpose of preset: `preset-kyma-guard-bot-github-identity` ?

File: prow/jobs/kyma/kyma-integration.yaml [hunk 3]
```diff
@@ -192,6 +194,7 @@ presubmits: # runs on PRs
     # following regexp won't start build if only Markdown files were changed
     run_if_changed: "^((resources\\S+|installation\\S+)(\\.[^.][^.][^.]+$|\\.[^.][^dD]$|\\.[^mM][^.]$|\\.[^.]$|/[^.]+$))"
     labels:
+      preset-build-pr: "true"
```
Comment (aszecowka) [inline prow/jobs/kyma/kyma-integration.yaml]: specify equivalent presets also for postsubmit and release jobs

File: prow/jobs/kyma/kyma-integration.yaml [hunk 4]
```diff
@@ -176,6 +177,7 @@ gke_xip_central_job_template: &gke_xip_central_job_template
           cpu: 80m
 
 gke_job_labels_template: &gke_job_labels_template
+  preset-kyma-guard-bot-github-token: "true"
```
Comment (aszecowka) [inline prow/jobs/kyma/kyma-integration.yaml]: describe your preset in `presets.md` file

File: development/tools/cmd/jobguard/main.go [hunk 5]
```diff
@@ -39,63 +39,67 @@ func main() {
 	time.Sleep(cfg.InitialSleepTime)
 
 	repoName := fmt.Sprintf("%s/%s", cfg.Status.Owner, cfg.Status.Repository)
-	jobsNumber, err := getNumberOfJobs(cfg.Prow.ConfigFile, cfg.Prow.JobsDirectory, repoName)
```
Comment (aszecowka) [inline development/tools/cmd/jobguard/main.go]: please describe briefly what to do if someone is going to introduce a new integration job. IMO good place can be `prow-jobs.md` file.

File: development/tools/jobs/kyma/kyma_integration_test.go (hunk 2 above)
Comment (mszostok) [reply development/tools/jobs/kyma/kyma_integration_test.go]: I just followed the previous approach. Right now it's not used anywhere but at least in context of the `GithubToken` you have information about its identity. Should be removed?

File: prow/scripts/cluster-integration/README.md [hunk 6]
```diff
@@ -17,6 +17,14 @@ The integration job is a pipeline that consists of multiple steps:
 - Test the Kyma installation.
 - Clean up all provisioned resources, also if you get an error.
 
+> **NOTE:** [Job Guard](./../../../development/tools/cmd/jobguard/README.md) must be execute at the beginning of each integration job that builds and installs Kyma witch changes from the pull-request.
```
Comment (kazydek) [inline prow/scripts/cluster-integration/README.md]:
```suggestion
> **NOTE:** Run [Job Guard](./../../../development/tools/cmd/jobguard/README.md) at the beginning of each integration job that builds and installs Kyma witch changes from the pull request.
```

File: prow/scripts/cluster-integration/README.md [hunk 7]
```diff
@@ -17,6 +17,14 @@ The integration job is a pipeline that consists of multiple steps:
 - Test the Kyma installation.
 - Clean up all provisioned resources, also if you get an error.
 
+> **NOTE:** Run [Job Guard](./../../../development/tools/cmd/jobguard/README.md) at the beginning of each integration job that builds and installs Kyma witch changes from the pull request.
```
Comment (kazydek) [inline prow/scripts/cluster-integration/README.md]:
```suggestion
> **NOTE:** Run [Job Guard](./../../../development/tools/cmd/jobguard/README.md) at the beginning of each integration job that builds and installs Kyma with changes from the pull request.
```

**Pull request merged**